9.0 pos access right

legalsylvain opened the pull request:
This module allow you to reduce the PoS rights for some users. For example, you can set that a user can not use discount option, or changing unit prices.
6 options are available : 

![new_groups](https://cloud.githubusercontent.com/assets/3407482/18231527/e74a80aa-72bb-11e6-8160-4d1c868b0091.png)

If you change PoS user during the session (by clicking on the name of the user), the right will be updated.

A message will be displayed, if the user try to use a unauthorized feature : 
![demo_error](https://cloud.githubusercontent.com/assets/3407482/18231549/5e3d59c6-72bc-11e6-9eb6-9cabe34effee.png)

Works on production since 2 months. Review welcome.

legalsylvain committed:
[ADD] new module 'pos_access_right'

legalsylvain committed:
[REF] improve description;

legalsylvain committed:
[ADD] fr translation;

legalsylvain committed:
[ADD] new blockage;

legalsylvain committed:
[ADD] fr translation;
[IMP] description;

legalsylvain committed:
[FIX]

legalsylvain committed:
[FIX] display;

mcassuto commented:
+1 (I have no rights, but cool feature !)

simahawk commented on pos_access_right/models/pos_config.py:
```diff
@@ -0,0 +1,88 @@
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+
+    group_pos_negative_qty = fields.Many2one(
```
aren't these fields supposed to have `_id` suffix?

legalsylvain replied:
You're right, I'll fix that.

simahawk replied:
if you want to keep field names shorter, you can drop `_pos_`... we know we are talking about "pos" things here ;)

legalsylvain committed:
[FIX] remove useless feature;[FIX] bug on multi order check

legalsylvain committed:
[REF] rename fields

legalsylvain committed:
[FIX] add missing author

legalsylvain committed:
[REF] improve fr translation

legalsylvain commented:
@phuctran1105 : I fixed the bug you mentioned. Could you test ?

@simahawk : I removed "_pos_" prefix and added "_id" suffix, according your remark.

Thanks for your review.

rafaelbn commented:
Please @legalsylvain could you rebase?

mslokhat commented:
Hello
my 2 cents (i already asked Yelizariev to add this in his module) :
add a setting to prevent the staff to make a payment. it can be useful for staff taking only orders and only a specific cashier can take the money
Rgds

legalsylvain commented:
@rafaelbn : rebase done.
@mslokhat : yes this feature could be integrated into this module, but I have not time for that.

regards.

rafaelbn commented:
Hi @legalsylvain , thanks. Tested in runbot 👍 

But please if possible add a section in user setting for this group. If not when mor modules installed it's not clear. Not blocking.

![2016-12-16_17-02-16](https://cloud.githubusercontent.com/assets/8736623/21269743/33a1ef26-c3b4-11e6-8356-51f68caced45.png)

![2016-12-16_17-10-20](https://cloud.githubusercontent.com/assets/8736623/21269749/3983f2ea-c3b4-11e6-81e6-c8a2fe7d1c98.png)

legalsylvain commented:
Hi @rafaelbn : indeed, it will be more easy to configure with a given "point of sale" group. But this group should be in the core module, and not in this one.
Alternatively, I can create a "Point Of Sale Access Right" Group if you want. 

What do you think ?

legalsylvain commented:
Well, let's go merging, and group refactoring could be done in a second time.

rafaelbn commented:
👍
